core/shared/src/main/scala/org/http4s/Header.scala

```diff
@@ -153,12 +153,12 @@ object Header {
 
     implicit def rawToRaw(h: Header.Raw): Header.ToRaw with Primitive =
       new Header.ToRaw with Primitive {
-        val values = List(h)
+        val values = h :: Nil
```

Comment:
Ideally the compiler should be doing the same thing, but Scala 3 is missing this optimization.
- https://github.com/lampepfl/dotty/issues/17035